File: lib/src/main/java/com/artufimtcev/sadapter/StrategyItem.java
package com.artufimtcev.sadapter;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;


public interface StrategyItem<VH extends RecyclerView.ViewHolder> {

	void onBindViewHolder(VH holder);
	VH createViewHolder(ViewGroup parent);
}

File: klib/src/main/java/com/artufimtcev/sadapter/StrategyItem.kt
package com.artufimtcev.sadapter

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

interface StrategyItem<VH : RecyclerView.ViewHolder> {

    fun onBindViewHolder(holder: VH)
    fun createViewHolder(parent: ViewGroup): VH
}
